### DISCUSSION TITLE
How to make a string parser with escapeable quotes?

### DISCUSSION THREAD

**Discussion by cAttte** [zesterer/chumsky#692] [Q&A] [answered]
hey! i've been ripping my hair out for a few hours now; i haven't been able to implement a quoted-string parser that allows for escaping quotes (so something like `"he\"llo"` → `Str(hel"lo)`) in an elegant way.

i've tried looking in the [sample JSON parser](https://github.com/zesterer/chumsky/blob/b748447b49b92716c4ec35288728496361f01c1b/examples/json.rs#L64) but (a) i don't know where the `to_slice()` method is supposed to come from and (b) there's also the `ignored()` which just turns the whole thing into a `Vec<()>`. so either this is really outdated or i just don't know what i'm doing. i've also tried to look in the other sample parsers but none of them try to implement this.

after several iterations, i've been able to get this working in a horrible 200-char line of code, but i feel like there has to be a better way. help would be really appreciated :)

**Comment by zesterer** [comment, marked as answer]
The `.to_slice()` method is part of chumsky: it gives you a parser whose output is a slice of the input that was accepted by the original parser. For example:

```rs
// Will yield "ab" as the output value
just('a').then('b').to_slice()
```

It might be that your confusion is due to using the `0.9` release (the latest stable release on crates.io), but the examples in the repository are written against `1.0` (which is still in alpha, although is nearing a stable release).

If you're starting a new project afresh I would strongly recommend using 1.0: it is 'the future' of chumsky, and it's much faster, more expressive, and will have support going forward. 0.9 is now considered legacy. If you want to persist with 0.9 though, you can check out the examples as they are [in the `legacy` branch](https://github.com/zesterer/chumsky/blob/legacy/examples/json.rs).

**Reply by cAttte** [reply]
ohh, so it wasn't the sample that was outdated but me! everything's working now--took me a bit to convert everything else to v1 but the guide is pretty good. thank you :)
